model/jpa/src/main/java/org/keycloak/models/jpa/UserAdapter.java

```diff
@@ -515,6 +517,11 @@ public void setServiceAccountClientLink(String clientInternalId) {
         user.setServiceAccountClientLink(clientInternalId);
     }
 
+    @Override
+    public SingleUserCredentialManager getUserCredentialManager() {
+        return session.getProvider(SingleUserCredentialManagerProvider.class).create(realm, this);
```

Comment:
Would this work if `SingleUserCredentialManagerProvider` was a map user credential manager?
I believe this should be a concrete implementation rather than provider lookup, i.e.
```suggestion
        return new DefaultSingleUserCredentialManager(realm, this, new DefaultSingleUserCredentialManagerStrategy(session, realm, user));
```